boards/DCM/Src/Io/Io_Imu_LSM6DS33.c

```diff
@@ -0,0 +1,189 @@
+    }
+
+    // Set Accelerometer to read at 416hz, max acceleration to +/-2g,
+    // anti-aliasing filter bandwidth at 400hz
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x60; // 0b0110_00_00
+        status       = Io_Imu_LSM6DS33_writeToImu(CTRL1_XL, &data, 1);
+    }
+
+    // Enable Gyroscope X,Y,Z axis
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x38; // 0b00_111_0_0_0
+        status       = Io_Imu_LSM6DS33_writeToImu(CTRL10_C, &data, 1);
+    }
+
+    // Set Gyroscope to read at 416hz, with max expected rotation of 250dps
+    // (degrees per second)
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x60; // 0b0110_00_0_0
+        status       = Io_Imu_LSM6DS33_writeToImu(CTRL2_G, &data, 1);
+    }
+
+    // Set IMU to trigger an interrupt on INT2 whenever there is new
+    // gyroscope or accelerometer data available
+    if (status == HAL_OK)
+    {
+        uint8_t data = 0x3; // 0b00000011
+        status       = Io_Imu_LSM6DS33_writeToImu(INT2_CTRL, &data, 1);
+    }
+
+    initialized = (status == HAL_OK);
+
+    return status;
+}
+
+bool Io_Imu_LSM6DS33_UpdateImuData()
+{
+    uint8_t           data[12];
+    HAL_StatusTypeDef status = HAL_ERROR;
+
+    if (initialized)
+    {
+        status = Io_Imu_LSM6DS33_readFromImu(DATA_OUT_STRT, data, 12);
+    }
+
+    uint8_t status_reg_val = 0;
+    Io_Imu_LSM6DS33_readFromImu(STATUS_REG, &status_reg_val, 1);
+
+    uint32_t current_time_in_ms = HAL_GetTick();
+
+    if (status == HAL_OK && status_reg_val != 0)
+    {
+        most_recently_received_data.received_time_ms = current_time_in_ms;
+
+        // This is how you can read data from the gyroscope. Note that these
+        // are raw values, please refer to the spec sheet (section 9.12) for
+        // information on how to convert it to meaningful information
+        // int16_t gyro_x = (data[1] << 8) + data[0];
+        // int16_t gyro_y = (data[3] << 8) + data[2];
+        // int16_t gyro_z = (data[5] << 8) + data[4];
+
+        int16_t raw_accel_x = (int16_t)((data[7] << 8) + data[6]);
+        int16_t raw_accel_y = (int16_t)((data[9] << 8) + data[8]);
+        int16_t raw_accel_z = (int16_t)((data[11] << 8) + data[10]);
+
+        most_recently_received_data.accel_x =
+            Io_Imu_LSM6DS33_convertIMUAccelerationToMetersPerSecondSquared(
+                raw_accel_x);
+        most_recently_received_data.accel_y =
+            Io_Imu_LSM6DS33_convertIMUAccelerationToMetersPerSecondSquared(
+                raw_accel_y);
+        most_recently_received_data.accel_z =
+            Io_Imu_LSM6DS33_convertIMUAccelerationToMetersPerSecondSquared(
+                raw_accel_z);
+
+        most_recently_received_data_valid = true;
+    }
+    else
+    {
+        most_recently_received_data_valid = false;
+    }
+
+    return ((status == HAL_OK) && most_recently_received_data_valid);
+}
+
+struct ImuData *Io_Imu_LSM6DS33_GetImuData()
```

Comment:
I'd rather have one function returning x acceleration, another returning y, and another returning z than this

Reply:
https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/678/files#r511185745

Reply:
Ok. It's still stored as ImuData in the IO code but the functions now return components.